### PR TITLE
Add hibernation calculation error details to webhooks

### PR DESCRIPTION
**Pull request by gabrieljackson**
The hibernation report sent to mattermost now includes calculation
error details to allow for quick investigation. The runtime
duration reported has also been cleaned up.

Addresses https://mattermost.atlassian.net/browse/MM-37421

### CODE REVIEW

File: cmd/fleet-controller/webhook.go [hunk 1]
```diff
@@ -53,13 +54,25 @@ Filters:
...
+func sendHibernateWebhook(webhookURL, runID, runtime, groupID, ownerID string, days, maxUsers, stableCount, hibernatedCount, skippedCount, errorCount int, errorDetails []string) error {
 	webhookText := fmt.Sprintf(
 		hibernateReportMessage,         // Text template
 		wrapInlineCode(runID), runtime, // Run data
 		days, maxUsers, wrapInlineCode(groupID), wrapInlineCode(ownerID), // Filters
 		stableCount, hibernatedCount, skippedCount, errorCount, // Results
 	)
+	if len(errorDetails) != 0 {
+		// Trim errors if necessary to prevent message bloat.
+		if len(errorDetails) > 10 {
```
Comment (Szymongib) [inline cmd/fleet-controller/webhook.go:71]: Maybe it would be worth it to make this limit configurable?

**Review by Szymongib — APPROVED** [review]
Very nice improvement, it might prevent weird things from sneaking by 👍

File: cmd/fleet-controller/webhook.go [hunk 2]
```diff
@@ -53,13 +54,25 @@ Filters:
 | Hibernation Calculation Errors | %d |
 `
 
-func sendHibernateWebhook(webhookURL, runID, runtime, groupID, ownerID string, days, maxUsers, stableCount, hibernatedCount, skippedCount, errorCount int) error {
+const hibernateReportErrorsSection = `
+#### Errors
+%s
+`
+
+func sendHibernateWebhook(webhookURL, runID, runtime, groupID, ownerID string, days, maxUsers, stableCount, hibernatedCount, skippedCount, errorCount int, errorDetails []string) error {
```
Comment (gigawhitlocks) [inline cmd/fleet-controller/webhook.go:62]: lol with 12 positional arguments maaaaaaaaybe it's time for an options struct for this function to receive

**Review by gigawhitlocks — APPROVED** [review]

File: cmd/fleet-controller/webhook.go (hunk 2 above)
Comment (gabrieljackson) [reply cmd/fleet-controller/webhook.go:62]: Not going to stop until we hit 100 inputs to be honest. 🚀

**Pull request merged**